test/native-promises.tap.js

```diff
@@ -1480,6 +1481,31 @@ test('Promise.resolve().catch().then()', function (t) {
     process.removeAsyncListener(listenerState.listener);
 });
 
+test('continue from unwrapped promise', function(t) {
+  var listener = addListner();
+
+  listener.currentName = 'resolve';
+  resolvedBeforeWrap.then(function(val) {
+    t.equal(val, 123, 'should match resolved value')
+    listener.currentName = '2nd resolve';
+    return 456
```

Comment:
We could propagate across this boundary (return -> 2nd then cb), haven't quite figured out the best way to do it yet though.